### ISSUE TITLE
No Display on Phillips TV

### ISSUE THREAD

**Issue by william-pyle** [rogro82/PiPup#9]
I have installed the beta from app store, so assume that means no extra permissions required.
However reviewing permissions on the app does show NO PERMISSIONS.

Submitting your exact example JSON via a test app, and I get a good 200 message back from the server, Not sure if a return of the same content is expected, but it sent me back a plain text of the JSON i submitted.  I did run top on the tv and saw that with each submission nl.rogro82.pipup did pop up in the top 10 briefly. So its so working the request.

MODEL:  Philips 4k TV
Version: 8.0.0
Android TV Core Services: 1.11.0-270169296-f
Google: 3.12.0.2701455
Google Services Framework: 8.0.0

If you need ANYTHING, or want anything special ran please advise.

FYI - I LOVE THIS IDEA!!!! Once this is all good, may fork with some MQTT extensions...

**Comment by gent99** [comment]
same here: Nothing is displayed on the TV, just the json is returned as plain text with http result 200. So if something is missing here, please let us know, what you need or we should do to test it

**Comment by lucasvdh** [comment]
Works fine for me on a Phillips TV.

**Comment by william-pyle** [comment]
lucasvdf, did you side load or get from Google play? and did you grant any permissions manually?

**Comment by gent99** [comment]
FYI: I get it from google play and did not grant any permissions via adb

**Comment by william-pyle** [comment]
gent99, that is what I did as well.

**Comment by william-pyle** [comment]
GOT IT WORKING!!!!
Ok, while everything says if you install via Google Play should be ok, its not setting allow on SYSTEM_ALERT_WINDOW.

I enabled developer options, enabled usb debugging. And because I didnt want to try to get a usb cable on the back of the tv mounted on the wall, I downloaded "Remote ADB Shell" from google play.  Put in the IP of my tv and connected right up.

ran:  appops set nl.rogro82.pipup SYSTEM_ALERT_WINDOW allow

Then reran the test with the client on my phone using the example JSON here.

It popped right up on my screen as advertised!!!!!

So while there is an issue with setting the permission in the Google Play installed version, it does work fine if manually done!!!!

Integrating this into my HomeAutomation stuff today!!!!!!

Gent99, if this also works for you, this can probably be closed once the Permission issue from Google play is at least acknowledge by rogro82.

**Comment by gent99** [comment]
No, even this did not work or me. I installed adb with homebrew on my mac with "brew cask install android-platform-tools". Then adb connect IP of my TV and then "adb shell appops set nl.rogro82.pipup SYSTEM_ALERT_WINDOW allow" And then use the example json with postman. Got a http 200 and returned the text of the PopupProps, but nothing is happening on the TV screen.

**Comment by dajojonl** [comment]
Sony bravia 65A1 tv  Android oreo 8
Developermode on tv with USB  debugging\foutopsporing turned on and on phone use remote adb Shell    
Type:
 appops set nl.rogro82.pipup SYSTEM_ALERT_WINDOW allow
And it Works :)
Thnx projecteleanor !

**Comment by gent99** [comment]
i got it working now: I updated the firmware of my philips to the latest version (based on android tv 8) and now it works

**Comment by lucasvdh** [comment]
@ProjectEleanor sorry for the late reply, but I feel it's still relevant. 

I installed pipup straight from the Play Store and didn't have to set or grant any permissions. This implies that there's a difference between tv's and how they handle what permissions can be set. I don't have the details of my tv at hand now but I'll post them later. Maybe it's possible to make a distinction between models/android versions which can set this permission when you install it from the store and which cant.

**Comment by william-pyle** [comment]
@lucasvdh you make a great point! I posted all my pics info originally, so yours would be very interesting to compare.

**Comment by lucasvdh** [comment]
Completely forgot to look this up, I'll do it tonight.

**Comment by lucasvdh** [comment]
Model: TPM171E
Version: 8.0.0

@ProjectEleanor how did you get the other hardware software info from your TV, is there an option or do you need to fetch it via adb?

**Comment by william-pyle** [comment]
@lucasvdh , go into settings, and scroll to about and scroll down. At least on mine... 😀

**Comment by hoasis2** [comment]
> how did you get the other hardware software info from your TV, is there an option or do you need to fetch it via adb?

[adb shell getprop](https://gist.github.com/Pulimet/5013acf2cd5b28e55036c82c91bd56d8)
